quinn-proto/src/connection.rs

```diff
@@ -604,20 +612,24 @@ impl Connection {
             for (&packet, info) in space.sent_packets.range(0..space.largest_acked_packet) {
                 if info.time_sent <= lost_send_time || packet <= lost_pn {
                     lost_packets.push(packet);
-                } else if self.loss_time == 0 {
-                    self.loss_time = info.time_sent + loss_delay;
                 } else {
-                    self.loss_time = cmp::min(self.loss_time, info.time_sent + loss_delay);
+                    let next_loss_time = info.time_sent + loss_delay;
+                    self.loss_time = Some(self.loss_time.map_or(next_loss_time, |loss_time| {
+                        cmp::min(loss_time, next_loss_time)
+                    }));
                 }
             }
 
             // OnPacketsLost
             if let Some(largest_lost) = lost_packets.last().cloned() {
                 let old_bytes_in_flight = self.in_flight.bytes;
-                largest_lost_time = cmp::max(
-                    largest_lost_time,
-                    space.sent_packets[&largest_lost].time_sent,
-                );
+                largest_lost_time =
+                    Some(largest_lost_time.map_or(
+                        space.sent_packets[&largest_lost].time_sent,
```

Comment:
This lookup might be expensive, so it would be best to extract it into a `let` binding rather than perform it twice when `largest_lost_time` is `Some(_)`.

Reply:
Fixed.